kratos/geometries/nurbs_curve_tessellation.h

```diff
@@ -0,0 +1,227 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Andreas Apostolatos
+//					 Tobias Teschemacher
+//					 Thomas Oberbichler
+//					
+//
+//  Ported from the ANurbs library (https://github.com/oberbichler/ANurbs)
+//
+
+#if !defined(KRATOS_NURBS_CURVE_TESSELLATION_H_INCLUDED )
+#define  KRATOS_NURBS_CURVE_TESSELLATION_H_INCLUDED
+
+#include "utilities/math_utils.h"
+#include "geometries/geometry.h"
+#include "geometries/nurbs_curve_geometry.h"
+
+#include "includes/ublas_interface.h"
+#include "containers/array_1d.h"
+
+namespace Kratos {
+
+// template <typename TCurve>
+template <int TWorkingSpaceDimension, class TContainerPointType>
+class NurbsCurveTessellation
+{
+public:
+
+    ///@name Type Definitions
+    ///@{
+
+    typedef Geometry<typename TContainerPointType::value_type> GeometryType;
+    typedef NurbsCurveGeometry<TWorkingSpaceDimension, TContainerPointType> NurbsCurveGeometryType;
+    typedef typename GeometryType::IndexType IndexType;
+    typedef typename GeometryType::SizeType SizeType;
+
+private:    // static methods
+    static double DistanceToLine(
+        const typename GeometryType::CoordinatesArrayType& Point, 
+        const typename GeometryType::CoordinatesArrayType& LineA,
+        const typename GeometryType::CoordinatesArrayType& LineB
+        )
+    {
+        typename GeometryType::CoordinatesArrayType vector_v = LineA - Point;
+        typename GeometryType::CoordinatesArrayType vector_u = LineB - LineA;
+
+        return MathUtils<double>::Norm(MathUtils<double>::CrossProduct(vector_v, vector_u)) / MathUtils<double>::Norm(vector_u);
+    }
+
+public:
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Conctructor for tessellation of a nurbs curve
+    NurbsCurveTessellation()
+    {
+    }
+
+    /** 
+    * @brief This method tessellates a curve and stores the tessellation in the class
+    * From ANurbs library (https://github.com/oberbichler/ANurbs)
+    * @param pGeometry Pointer to the geometry
+    * @param PolynomialDegree The polynomial degree of the curve
+    * @param DomainInterval The curve interval which is to be tessellated
+    * @param KnotSpanIntervals The knot span intervals laying in the DomainInterval
+    * @param Tolerance Tolerance for the choral error
+    * @see ComputeTessellation
+    */
+    void Tessellate(
+        typename GeometryType::Pointer pGeometry, 
+        int PolynomialDegree,
+        NurbsInterval DomainInterval,
+        std::vector<NurbsInterval> KnotSpanIntervals,
+        const double Tolerance)
+    {
+        mTesselation = ComputeTessellation<TWorkingSpaceDimension>(
```

Comment:
So the idea here is to have a static member function which can be called without creating an object of the class (in case we don't want to store the tessellation) and a member function that calls the static one and stores the tessellation. If you have anything against the naming feel free to propose alternatives and I will change them accordingly.